Jantimizer/Histograms/IDepthHistogramManager.cs

```diff
@@ -7,9 +7,10 @@
 using Histograms.Models;
 
 namespace Histograms
-{
+{
+    public delegate int DepthCalculator(long uniqueValueCount, long totalValueCount);
```

Comment:
Is there any good reason to use delegates here? Why all of a sudden such a c++ way of writing code?
Just set the `DepthCalculators` to use a common interface, and pass an object around instead.